protocol-units/bridge/chains/movement/src/utils.rs

```diff
@@ -113,22 +111,40 @@ pub async fn send_and_confirm_aptos_transaction(
 	let transaction_factory = TransactionFactory::new(ChainId::new(state.chain_id))
 		.with_gas_unit_price(100)
 		.with_max_gas_amount(GAS_UNIT_LIMIT);
-	debug!("Transaction factory created with chain_id = {}", state.chain_id);
+	let latest_account_info = rest_client.get_account(signer.address()).await?;
+	let account = latest_account_info.into_inner();  
+	let latest_sequence_number = account.sequence_number;	
 
-	let signed_tx = signer.sign_with_transaction_builder(transaction_factory.payload(payload));
-	debug!("Transaction signed: {:?}", signed_tx);
+	let raw_tx = transaction_factory
+		.payload(payload)
+		.sender(signer.address())
+		.sequence_number(latest_sequence_number)
+		.build();
+
+	let signed_tx = signer.sign_transaction(raw_tx);
 
 	let response = rest_client
 		.submit_and_wait(&signed_tx)
 		.await
 		.map_err(|e| {
-		debug!("Transaction submission failed: {:?}", e);
 		anyhow::anyhow!(e.to_string())
 		})?
 		.into_inner();
+
+	match &response {
+		Transaction::UserTransaction(user_txn) => {
+			assert!(
+			user_txn.info.success,
+			"Transaction failed with status: {}",
+			user_txn.info.vm_status
+			);
+		},
+		_ => panic!("Expected a UserTransaction, but got a different transaction type."),
```

Comment:
Lets Return the the `Err` variant of the Result type here rather than `panic!`, so that if the  _=> arm is hit, the service will not crash but gracefully handle the error and return that to the calling context.